### PR TITLE
feat: add attachment field

### PR DESCRIPTION
**Pull request by MarvinJanssen**
## Description

Added an optional attachment field to `TxBase` for use with the Web Extension. (Is `TxBase` the right spot for it since an attachment is submitted alongside a transaction?)

## Type of Change
- [x] New feature
- [ ] Bug fix
- [ ] API reference/documentation update
- [ ] Other

## Does this introduce a breaking change?
No.

## Are documentation updates required?
Yes. TBD.

## Checklist
- [ ] Code is commented where needed
- [ ] Unit test coverage for new or modified code paths
- [ ] `yarn lerna run test` passes
- [ ] Changelog is updated
- [ ] Tag 1 of @hstove or @kyranjamie or @aulneau for review

### CODE REVIEW

File: packages/connect/package.json [hunk 1]
```diff
@@ -20,7 +20,7 @@
   "unpkg": "dist/bundle.js",
   "dependencies": {
     "@stacks/auth": "^1.0.2",
-    "@stacks/connect-ui": "^5.0.5",
+    "@stacks/connect-ui": "^2.17.19",
```
Comment (hstove) [inline packages/connect/package.json:23]: What was the reasoning for changing this? We want to keep this synced with the version of `connect/ui`

**Review by hstove — APPROVED** [review]
I think this is right! Will want to setup a web wallet PR and test with that before merging this.

File: packages/connect/package.json (hunk 1 above)
Comment (MarvinJanssen) [reply packages/connect/package.json:23]: Oh curious, that happened when I ran my install. Will undo.

**Comment by markmhx** [comment]
Related: https://github.com/blockstack/stacks-wallet-web/pull/1132

**Comment by aulneau** [comment]
thanks @MarvinJanssen, i need to close this and do it in a different pr, https://github.com/blockstack/connect/pull/113

**Pull request closed without merging**